src/main/java/net/sf/jsqlparser/statement/select/Join.java

```diff
@@ -278,49 +294,58 @@ public void setJoinWindow(KSQLJoinWindow joinWindow) {
         this.joinWindow = joinWindow;
     }
 
-    @Override
     @SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.NPathComplexity"})
-    public String toString() {
+    public StringBuilder appendTo(StringBuilder builder) {
         if (isSimple() && isOuter()) {
-            return "OUTER " + rightItem;
+            builder.append("OUTER ").append(rightItem);
         } else if (isSimple()) {
-            return "" + rightItem;
+            builder.append(rightItem);
         } else {
-            String type = "";
-
             if (isRight()) {
-                type += "RIGHT ";
+                builder.append("RIGHT ");
             } else if (isNatural()) {
-                type += "NATURAL ";
+                builder.append("NATURAL ");
             } else if (isFull()) {
-                type += "FULL ";
+                builder.append("FULL ");
             } else if (isLeft()) {
-                type += "LEFT ";
+                builder.append("LEFT ");
             } else if (isCross()) {
-                type += "CROSS ";
+                builder.append("CROSS ");
             }
 
             if (isOuter()) {
-                type += "OUTER ";
+                builder.append("OUTER ");
             } else if (isInner()) {
-                type += "INNER ";
+                builder.append("INNER ");
             } else if (isSemi()) {
-                type += "SEMI ";
+                builder.append("SEMI ");
             }
 
             if (isStraight()) {
-                type = "STRAIGHT_JOIN ";
+                builder.append("STRAIGHT_JOIN ");
             } else if (isApply()) {
-                type += "APPLY ";
+                builder.append("APPLY ");
             } else {
-                type += "JOIN ";
+                builder.append("JOIN ");
             }
-
-            return type + rightItem + ((joinWindow != null) ? " WITHIN " + joinWindow : "")
-                    + ((onExpression != null) ? " ON " + onExpression + "" : "")
-                    + PlainSelect.getFormatedList(usingColumns, "USING", true, true);
+            
+            builder.append(rightItem).append((joinWindow != null) ? " WITHIN " + joinWindow : "");
+        }
+        
+        
+        for (Expression onExpression: onExpressions) {
+            builder.append(" ON ").append(onExpression);
+        }
+        if (usingColumns.size()>0) {
+            builder.append(PlainSelect.getFormatedList(usingColumns, "USING", true, true));
         }
 
+        return builder;
+    }
+
+    @Override
+    public String toString() {
+        return appendTo(new StringBuilder()).toString();
```

Comment:
Again. Whats the improvement of this type of generating a StringBuilder and populating it? IMHO this is no improvement over this older toString approach.

Reply:
Good Morning. Thank you for your feedback.
My thought process was like this:

1. I have had to touch `toString()` because of the new Loop through the ON-Expressions
2. String Addition `+=` in Loops is still considered bad style and will be called out by many style checkers, because the Compiler **won't** substitute the concatenation by a StringBuilder
3. providing an `appendTo()` method consistently for all Statements and Expressions would eventually allow the Deparser to work with one single StringBuilder only (imitated at the top). Right many Statements and Expressions initiate and dispose their own StringBuilder putting useless load onto the GC (think about deep nested expressions especially).
4. I do not see any disadvantages of this pattern. `toString()` still works without any caveats. We just delegate the logic into a special method. You did the same for many other Statements and Expressions already -- just not calling it `appendTo()` and not applying a systematic pattern.

What counter arguments do you see? Why was the String concatenation better, especially with the newly needed Loops?

Cheers.

Reply:
I have removed the `appendTo()` method and re-implemented the traditional `toString()` as you obviously do not like this approach.